src/main/java/uk/gov/hmcts/reform/sscscorbackend/util/UpdateSubscription.java

```diff
@@ -0,0 +1,53 @@
+package uk.gov.hmcts.reform.sscscorbackend.util;
+
+import lombok.extern.slf4j.Slf4j;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.domain.Subscription;
+import uk.gov.hmcts.reform.sscs.ccd.domain.Subscriptions;
+
+@Slf4j
+public class UpdateSubscription {
+
+    private static final String YES = "Yes";
+    private static final String NO = "No";
+
+    private UpdateSubscription() {
+        // Empty
+    }
+
+    public static void updateSubscription(SscsCaseData existingCcdCaseData,
+                                   SubscriptionUpdate subscriptionUpdate,
+                                   String lastLoggedIntoMya) {
+
+        Subscriptions existingSubscriptions = existingCcdCaseData.getSubscriptions();
+
+        Subscription existingSubscription = subscriptionUpdate.getSubscription(existingSubscriptions);
+
+        Subscription updatedSubscription =
+                keepExistingSubscribedSubscriptions(existingSubscription, lastLoggedIntoMya);
+
+        existingSubscriptions = subscriptionUpdate.updateExistingSubscriptions(updatedSubscription);
+
+        existingCcdCaseData.setSubscriptions(existingSubscriptions);
+    }
+
+    private static Subscription keepExistingSubscribedSubscriptions(Subscription exisitingSubscription,
+                                                                    String lastLoggedIntoMya) {
+        return Subscription.builder()
```

Comment:
More of a question, but do you really have to rebuild the subscription here? Can't you just use the existingCcdCaseData object and overwrite the last logged in value?

Reply:
We can not directly update the Subscription as all properties are read only. If we want to set the property then it would be created new subscription from existing existingsubscription

Reply:
Most other fields in ccd data model are read and write access. It should be a case of just changing the lombok type between data and value (forget which way round it is now). Up to you, but seems like unnecessary logic which could get broken easily if a new field was added to the subscription object. Let me know what you want to do.

Reply:
Ok, let me check and refactor as you suggested. I have done similar to case loader https://github.com/hmcts/sscs-case-loader/blob/master/src/main/java/uk/gov/hmcts/reform/sscs/services/ccd/UpdateSubscription.java